pkg/providers/v1/aws.go

```diff
@@ -1668,7 +1668,22 @@ func (c *Cloud) NodeAddressesByProviderID(ctx context.Context, providerID string
 		if eni == nil || err != nil {
 			return nil, err
 		}
-		return getNodeAddressesForFargateNode(aws.StringValue(eni.PrivateDnsName), aws.StringValue(eni.PrivateIpAddress)), nil
+
+		var addresses []v1.NodeAddress
+
+		// Assign NodeInternalIP based on IP family
+		for _, family := range c.cfg.Global.NodeIPFamilies {
+			switch family {
+			case "ipv4":
+				nodeAddresses := getNodeAddressesForFargateNode(aws.StringValue(eni.PrivateDnsName), aws.StringValue(eni.PrivateIpAddress))
+				addresses = append(addresses, nodeAddresses...)
+			case "ipv6":
+				internalIPv6Address := eni.Ipv6Addresses[0].Ipv6Address
```

Comment:
Even if the NodeIPFamily is set to ipv6, I wonder if its possible for the ENI to not have IPv6 addresses due to a misconfiguration or something.  Can we check to make sure the addresses exist and handle a failure case before referencing the first item in the list here?

Reply:
Ack, added a check to make sure Ipv6Addresses is not missing or empty

Reply:
Valid point for that use case, I have updated to log the error and return the existing addresses